src/passes/SimplifyLocals.cpp

```diff
@@ -411,6 +411,13 @@ struct SimplifyLocals
     if (set->isTee()) {
       return false;
     }
+    // We cannot move expressions containing exnref.pops that are not enclosed
+    // in 'catch', because 'exnref.pop' should follow right after 'catch'.
+    if (EffectAnalyzer(
```

Comment:
how about checking for the exceptions feature here, to avoid computing all side effects when it is disabled?

Reply:
Done